storage/adapters/pgp_adapter.js

```diff
@@ -0,0 +1,194 @@
+			},
+			error: () => {
+				this.emit(this.EVENT_ERROR);
+			},
+			disconnect: () => {
+				this.emit(this.EVENT_DISCONNECT);
+			},
+		};
+
+		this.pgp = pgpLib(this.pgpOptions);
+		this.db = undefined;
+	}
+
+	/**
+	 * @return {Promise}
+	 */
+	connect() {
+		if (monitor.isAttached()) {
+			monitor.detach();
+		}
+
+		const monitorOptions = {
+			error: (error, e) => {
+				this.logger.error(error);
+
+				// e.cn corresponds to an object, which exists only when there is a connection related error.
+				// https://vitaly-t.github.io/pg-promise/global.html#event:error
+				if (e.cn) {
+					process.emit('cleanup', new Error('DB Connection Error'));
+				}
+			},
+		};
+		monitor.attach(
+			{ ...this.pgpOptions, ...monitorOptions },
+			this.options.logEvents
+		);
+		monitor.setLog((msg, info) => {
+			this.logger.log(info.event, info.text);
+			info.display = false;
+		});
+		monitor.setTheme('matrix');
+
+		this.options.user = this.options.user || process.env.USER;
+
+		this.pgp.end();
+		this.db = this.pgp(this.options);
+
+		// As of the nature of pg-promise the connection is acquired either a query is started to execute.
+		// So to actually verify the connection works fine
+		// based on the provided options, we need to test it by acquiring
+		// the connection a manually
+
+		let connectionObject = null;
+
+		return this.db
+			.connect()
+			.then(co => {
+				connectionObject = co;
+				return Promise.resolve(true);
+			})
+			.finally(() => {
+				if (connectionObject) {
+					connectionObject.done();
+				}
+				return Promise.resolve(true);
+			});
+	}
+
+	disconnect() {
+		this.logger.info('Disconnecting');
+		if (monitor.isAttached()) {
+			monitor.detach();
+		}
+	}
+
+	/**
+	 * Execute an SQL file
+	 *
+	 * @param {string} file
+	 * @param {Object} params
+	 * @param {Object} options
+	 * @param {Number} [options.expectedResult]
+	 * @param {Object} tx
+	 * @return {*}
+	 */
+	executeFile(file, params = {}, options = {}, tx) {
+		return (tx || this.db)[resultCountToMethodMap[options.expectedResult]](
+			file,
+			params
+		);
+	}
+
+	/**
+	 * Execute an SQL file
+	 *
+	 * @param {string} sql
+	 * @param {Object} params
+	 * @param {Object} options
+	 * @param {Number} [options.expectedResult]
+	 * @param {Object} tx
+	 * @return {*}
+	 */
+	execute(sql, params = {}, options = {}, tx) {
+		return (tx || this.db)[resultCountToMethodMap[options.expectedResult]](
+			sql,
+			params
+		);
+	}
+
+	transaction(name, cb, tx) {
+		return (tx || this.db).tx(name, cb);
+	}
+
+	task(name, cb, tx) {
+		return (tx || this.db).task(name, cb);
+	}
+
+	loadSQLFile(filePath) {
+		const fullPath = path.join(this.sqlDirectory, filePath); // Generating full path;
+
+		const options = {
+			minify: true, // Minifies the SQL
+		};
+
+		const qf = new QueryFile(fullPath, options);
+
+		if (qf.error) {
+			console.error(qf.error); // Something is wrong with our query file
+			process.exit(1); // Exit the process with fatal error
```

Comment:
I think deciding on `process.exit` at this low level is too dangerous.
Prefer just to throw an error